crates/deltalake-core/src/operations/write.rs

```diff
@@ -415,12 +476,31 @@ impl std::future::IntoFuture for WriteBuilder {
                 this.write_batch_size,
                 this.writer_properties,
                 this.safe_cast,
+                this.overwrite_schema,
             )
             .await?;
             actions.extend(add_actions.into_iter().map(Action::Add));
 
             // Collect remove actions if we are overwriting the table
             if matches!(this.mode, SaveMode::Overwrite) {
+                // Update metadata with new schema
+                let table_schema = this
+                    .snapshot
+                    .physical_arrow_schema(this.log_store.object_store().clone())
+                    .await
+                    .or_else(|_| this.snapshot.arrow_schema())
+                    .unwrap_or(schema.clone());
+
+                if schema != table_schema {
```

Comment:
Do the schemas here have Eq or PartialEq on them? This might not be as straight forward as this.

Reply:
It has PartialEq on them. This piece was also reused from src/lib.rs in the python module